// @leonardo-ai/sdk/dist/sdk/models/shared/custommodeltype.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomModelType = void 0;
/**
 * The category the most accurately reflects the model.
 */
var CustomModelType;
(function (CustomModelType) {
    CustomModelType["General"] = "GENERAL";
    CustomModelType["Buildings"] = "BUILDINGS";
    CustomModelType["Characters"] = "CHARACTERS";
    CustomModelType["Environments"] = "ENVIRONMENTS";
    CustomModelType["Fashion"] = "FASHION";
    CustomModelType["Illustrations"] = "ILLUSTRATIONS";
    CustomModelType["GameItems"] = "GAME_ITEMS";
    CustomModelType["GraphicalElements"] = "GRAPHICAL_ELEMENTS";
    CustomModelType["Photography"] = "PHOTOGRAPHY";
    CustomModelType["PixelArt"] = "PIXEL_ART";
    CustomModelType["ProductDesign"] = "PRODUCT_DESIGN";
    CustomModelType["Textures"] = "TEXTURES";
    CustomModelType["UiElements"] = "UI_ELEMENTS";
    CustomModelType["Vector"] = "VECTOR";
})(CustomModelType = exports.CustomModelType || (exports.CustomModelType = {}));
