// @leonardo-ai/sdk/dist/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Leonardo = exports.SDKConfiguration = exports.ServerList = void 0;
var utils = __importStar(require("../internal/utils"));
var dataset_1 = require("./dataset");
var generation_1 = require("./generation");
var initimage_1 = require("./initimage");
var model_1 = require("./model");
var shared = __importStar(require("./models/shared"));
var user_1 = require("./user");
var variation_1 = require("./variation");
var axios_1 = __importDefault(require("axios"));
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = [
    /**
     * Leonardo.Ai API server
     */
    "https://cloud.leonardo.ai/api/rest/v1",
];
var SDKConfiguration = /** @class */ (function () {
    function SDKConfiguration(init) {
        this.language = "typescript";
        this.openapiDocVersion = "v1.0.0";
        this.sdkVersion = "1.39.2";
        this.genVersion = "2.39.8";
        Object.assign(this, init);
    }
    return SDKConfiguration;
}());
exports.SDKConfiguration = SDKConfiguration;
/**
 * Rest Endpoints: Leonardo.Ai API OpenAPI specification.
 */
var Leonardo = /** @class */ (function () {
    function Leonardo(props) {
        var _a, _b;
        var serverURL = props === null || props === void 0 ? void 0 : props.serverURL;
        var serverIdx = (_a = props === null || props === void 0 ? void 0 : props.serverIdx) !== null && _a !== void 0 ? _a : 0;
        if (!serverURL) {
            serverURL = exports.ServerList[serverIdx];
        }
        var defaultClient = (_b = props === null || props === void 0 ? void 0 : props.defaultClient) !== null && _b !== void 0 ? _b : axios_1.default.create({ baseURL: serverURL });
        var securityClient = defaultClient;
        if (props === null || props === void 0 ? void 0 : props.security) {
            var security = props.security;
            if (!(props.security instanceof utils.SpeakeasyBase)) {
                security = new shared.Security(props.security);
            }
            securityClient = utils.createSecurityClient(defaultClient, security);
        }
        this.sdkConfiguration = new SDKConfiguration({
            defaultClient: defaultClient,
            securityClient: securityClient,
            serverURL: serverURL,
        });
        this.dataset = new dataset_1.Dataset(this.sdkConfiguration);
        this.generation = new generation_1.Generation(this.sdkConfiguration);
        this.initImage = new initimage_1.InitImage(this.sdkConfiguration);
        this.model = new model_1.Model(this.sdkConfiguration);
        this.user = new user_1.User(this.sdkConfiguration);
        this.variation = new variation_1.Variation(this.sdkConfiguration);
    }
    return Leonardo;
}());
exports.Leonardo = Leonardo;
