// @leonardo-ai/sdk/dist/sdk/models/operations/getgenerationsbyuserid.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetGenerationsByUserIdResponse = exports.GetGenerationsByUserId200ApplicationJSON = exports.GetGenerationsByUserId200ApplicationJSONGenerations = exports.GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages = exports.GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric = exports.GetGenerationsByUserIdRequest = void 0;
var utils_1 = require("../../../internal/utils");
var shared = __importStar(require("../shared"));
var class_transformer_1 = require("class-transformer");
var GetGenerationsByUserIdRequest = /** @class */ (function (_super) {
    __extends(GetGenerationsByUserIdRequest, _super);
    function GetGenerationsByUserIdRequest() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "pathParam, style=simple;explode=false;name=userId" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserIdRequest.prototype, "userId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=limit" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserIdRequest.prototype, "limit", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=offset" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserIdRequest.prototype, "offset", void 0);
    return GetGenerationsByUserIdRequest;
}(utils_1.SpeakeasyBase));
exports.GetGenerationsByUserIdRequest = GetGenerationsByUserIdRequest;
/**
 * columns and relationships of "generated_image_variation_generic"
 */
var GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric = /** @class */ (function (_super) {
    __extends(GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric, _super);
    function GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "status" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric.prototype, "status", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "transformType" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric.prototype, "transformType", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "url" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric.prototype, "url", void 0);
    return GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric;
}(utils_1.SpeakeasyBase));
exports.GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric = GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric;
/**
 * columns and relationships of "generated_images"
 */
var GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages = /** @class */ (function (_super) {
    __extends(GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages, _super);
    function GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({
            elemType: GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric,
        }),
        (0, class_transformer_1.Expose)({ name: "generated_image_variation_generics" }),
        (0, class_transformer_1.Type)(function () {
            return GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImagesGeneratedImageVariationGeneric;
        }),
        __metadata("design:type", Array)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages.prototype, "generatedImageVariationGenerics", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "likeCount" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages.prototype, "likeCount", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "nsfw" }),
        __metadata("design:type", Boolean)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages.prototype, "nsfw", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "url" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages.prototype, "url", void 0);
    return GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages;
}(utils_1.SpeakeasyBase));
exports.GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages = GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages;
/**
 * columns and relationships of "generations"
 */
var GetGenerationsByUserId200ApplicationJSONGenerations = /** @class */ (function (_super) {
    __extends(GetGenerationsByUserId200ApplicationJSONGenerations, _super);
    function GetGenerationsByUserId200ApplicationJSONGenerations() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "createdAt" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "createdAt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({
            elemType: GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages,
        }),
        (0, class_transformer_1.Expose)({ name: "generated_images" }),
        (0, class_transformer_1.Type)(function () { return GetGenerationsByUserId200ApplicationJSONGenerationsGeneratedImages; }),
        __metadata("design:type", Array)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "generatedImages", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "guidanceScale" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "guidanceScale", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "imageHeight" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "imageHeight", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "imageWidth" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "imageWidth", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "inferenceSteps" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "inferenceSteps", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "initStrength" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "initStrength", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "modelId" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "modelId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "negativePrompt" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "negativePrompt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "presetStyle" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "presetStyle", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "prompt" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "prompt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "public" }),
        __metadata("design:type", Boolean)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "public", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "scheduler" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "scheduler", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "sdVersion" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "sdVersion", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "seed" }),
        __metadata("design:type", Number)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "seed", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "status" }),
        __metadata("design:type", String)
    ], GetGenerationsByUserId200ApplicationJSONGenerations.prototype, "status", void 0);
    return GetGenerationsByUserId200ApplicationJSONGenerations;
}(utils_1.SpeakeasyBase));
exports.GetGenerationsByUserId200ApplicationJSONGenerations = GetGenerationsByUserId200ApplicationJSONGenerations;
/**
 * Responses for GET /generations/user/{userId}
 */
var GetGenerationsByUserId200ApplicationJSON = /** @class */ (function (_super) {
    __extends(GetGenerationsByUserId200ApplicationJSON, _super);
    function GetGenerationsByUserId200ApplicationJSON() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ elemType: GetGenerationsByUserId200ApplicationJSONGenerations }),
        (0, class_transformer_1.Expose)({ name: "generations" }),
        (0, class_transformer_1.Type)(function () { return GetGenerationsByUserId200ApplicationJSONGenerations; }),
        __metadata("design:type", Array)
    ], GetGenerationsByUserId200ApplicationJSON.prototype, "generations", void 0);
    return GetGenerationsByUserId200ApplicationJSON;
}(utils_1.SpeakeasyBase));
exports.GetGenerationsByUserId200ApplicationJSON = GetGenerationsByUserId200ApplicationJSON;
var GetGenerationsByUserIdResponse = /** @class */ (function (_super) {
    __extends(GetGenerationsByUserIdResponse, _super);
    function GetGenerationsByUserIdResponse() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", String)
    ], GetGenerationsByUserIdResponse.prototype, "contentType", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", Number)
    ], GetGenerationsByUserIdResponse.prototype, "statusCode", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", Object)
    ], GetGenerationsByUserIdResponse.prototype, "rawResponse", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", GetGenerationsByUserId200ApplicationJSON)
    ], GetGenerationsByUserIdResponse.prototype, "getGenerationsByUserId200ApplicationJSONObject", void 0);
    return GetGenerationsByUserIdResponse;
}(utils_1.SpeakeasyBase));
exports.GetGenerationsByUserIdResponse = GetGenerationsByUserIdResponse;
